addon/states.js

```diff
@@ -164,24 +164,23 @@ FragmentRootState = wireState(FragmentRootState, null, 'root');
 export default FragmentRootState;
 
 export function fragmentDidDirty(record, key, fragment) {
-  if (!get(record, 'isDeleted')) {
```

Comment:
This can get called while the record is still being initialized, which causes the `DS.Model#currentState` to be overwritten with a new descriptor which then triggers an assertion in the `DS.Model#init` method. I am not sure what the consequences of accessing the currentState directly could be. Can we expect it to always be the canonical state?